src/components/BudgetController.js

```diff
@@ -0,0 +1,64 @@
+import React,{useState} from 'react'
+
+export default function BudgetController({Categorys,CategoryAdd,ItemAdd}) {
+    const [newCategoryInput,setNewCategoryInput]=useState('');// 카테고리 생성용 state
+    const [currentItem,setItem]=useState({
+            itemCategory:'',
+            itemUsage:'',
+            itemCost:0
+    });//한번에 State 관리 하는 방법 생각해 보기
+    
+    const newCategoryOnchange=(e)=>{
+        setNewCategoryInput(e.target.value);
+    };
+
+    
+    return (
+        <div style={{backgroundColor:'#192F5E',height:'35%',width:'90%',marginLeft:"auto",marginRight: 'auto',borderRadius:'5px',marginTop:'3rem'}}>
+            <span>ADD category: <input type='text' value={newCategoryInput} onChange={(e)=>{newCategoryOnchange(e)}} onKeyPress={(e)=>{if(e.charCode===13){CategoryAdd(newCategoryInput);setNewCategoryInput('')}}}></input> <button onClick={()=>{CategoryAdd(newCategoryInput);setNewCategoryInput('')}}>ADD</button></span>
+            <hr/>
+            <div>
+                <span>
+                        <label htmlFor="selectCategory">Category : </label>
+                        <select id='selectCategory' onChange={(e)=>setItem({...currentItem,itemCategory:e.target.value})} value={currentItem.itemCategory}>
+                            <option key='default'>선택</option>
+                           {Categorys&&Categorys.map((category,index)=>{return(<option key={category.category}>{category.category}</option>)})}
+                        </select>
+                </span>
+                    <hr/>
+                <label htmlFor='usage'> Usage : </label>
+                <input id='usage' type='text' value={currentItem.itemUsage} onChange={(e)=>setItem({...currentItem,itemUsage:e.target.value})}></input>
+                    <hr/>
+                <label>Cost : </label>
+                <input id='Cost' type='text' value={currentItem.itemCost}
+                    onKeyPress={(e)=>{
+                            if(e.charCode===13){
+                                if(ItemAdd(currentItem)===-1)return;
+                                setItem(()=>{return{itemCategory:'',itemUsage:'',itemCost:0}})
+                            }
+                            else{
+                                return;
+                            
+                            }
+                        }
+                    } 
+                    onChange={(e)=>{
+                        if(isNaN(e.target.value)){
```

Comment:
https://stackoverflow.com/questions/33164725/confusion-between-isnan-and-number-isnan-in-javascript
use `Number.isNaN` instead